### PR TITLE
Switch to use Ghost-SDK tags helper

### PR DESCRIPTION
**Pull request by ErisDS**
~~only works when using `yarn link` 🤔 ~~

The original version of this PR used a react helper from @tryghost/helpers-react, but I'm not sure that concept is going to be worthwhile. If we want to have react components for gatsby they need to use gatsby-link, whereas a normal react helper would use react-router-dom's link.

I'm not sure there's an easy way to make a generic component, or if we really ought to have `helpers-gatsby` rather than `helpers-react` cc @JohnONolan 

For now, I've moved my attempt at a react component into this repo in place of the existing tags component, which uses the ghost tags helper to do filtering, limiting and so on.

I think it works.

### CODE REVIEW

**Comment by ErisDS** [comment]
@AileenCGN I've updated this PR 

@JohnONolan I don't think having generic react components is gonna work without quite a bit more playing (see PR desc). I think I will focus on having a more thorough suite of tools in @tryghost/helpers, and we can think about use cases for either @tryghost/helpers-react or @tryghost/helpers-gatsby when we get a bit further down the road.

File: src/components/common/Tags.js [hunk 1]
```diff
@@ -1,122 +1,98 @@
...
+*   - prefix [optional, default "", sets a prefix to appear before the tags]
+*   - suffix [optional, default "", sets a suffix to appear after the tags]
+*   - classes [optional, default """, classNames used for the html tags]
+*   - separatorClasses [optional, default "", classNames used for the html separator tags]
+*   - prefixClasses [optional, default "", classNames used for the html prefix tags]
+*   - suffixClasses [optional, default "", classNames used for the html suffix tags]
+*   - linkClasses [optional, default "", classNames used for the html link tags]
+*   - autolink [optional, default true, whether to convert tags to links]
+*   - permalink [optional, default "/:slug.", the pattern used for links]
+*   - visibility [optional, default "all", the pattern used to control the if internal tags should be included]
```
Comment (aileen) [inline src/components/common/Tags.js]: @ErisDS I added the visibility attribute, but it doesn't work as expected for me. I tried
- `public` → renders nothing
- `all` → renders everything
- `internal` → renders nothing

File: src/components/common/Tags.js [hunk 2]
```diff
@@ -1,122 +1,98 @@
...
-
-                    if (separator && i < tags.length - 1) {
-                        output.push(<span className={props.separatorClasses} key={`sep-${tag.slug}`}>{separator}</span>)
-                    }
-                } else {
-                    // Render tags as span elements
-                    output.push(<span className={props.classes} key={tag.slug}>{tag.name}</span>)
+    if (props.separator) {
+        opts.separator = React.isValidElement(props.separator) ? props.separator :
+            <span className={props.separatorClasses}>{props.separator}</span>
```
Comment (aileen) [inline src/components/common/Tags.js]: There are two problems here:
1. When I want **no** separator, I'll pass `separator=""`, which will then still add the default `", "` separator [here](https://github.com/TryGhost/Ghost-SDK/blob/master/packages/helpers/tags.js#L12):
image

2. Adding a separator (even if it's a `" "` to workaround problem no. 1), but having more than one tag will cause a problem with the required `key` attributes for React children:

image

This is fine for `prefix` and `suffix` as these get only appended once.

File: src/components/common/Tags.js [hunk 3]
```diff
@@ -1,122 +1,98 @@
...
-    } else {
-        // We have more than one tag left
-        if (tags.length > 1) {
-            tags = tags.map(tag => tag.name)
-            output.push(tags.join(`${separator}`))
-        } else {
-            output.push(tags.name)
-        }
+    opts.fn = function process(tag) {
+        let tagLink = props.permalink
```
Comment (aileen) [inline src/components/common/Tags.js]: We're getting linting errors here:

```
/Users/aileen/code/docs/src/components/common/Tags.js
  49:29  error  'permalink' is missing in props validation    react/prop-types
  52:22  error  'autolink' is missing in props validation     react/prop-types
  53:36  error  'classes' is missing in props validation      react/prop-types
  54:53  error  'linkClasses' is missing in props validation  react/prop-types
```

**Review by aileen — CHANGES_REQUESTED** [review]
`separator` and `visibility` don't work as expected. See detailed comments

File: src/components/tutorials/TutorialCard.js [hunk 4]
```diff
@@ -26,7 +24,14 @@ const TutorialCard = ({ post, className, section }) => {
             <footer className="pt2 mt6 flex justify-between items-center">
                 <div className="flex items-center">
                     {post.featured ? <span className="bg-tutorial-color pa1 f-supersmall fw5 dib measure-0-2 mr2 white br2 pl3 pr3">Featured</span> : null}
-                    {tags ? tags.map((tag, i) => <span key={i} className="bg-midgrey dib word-nowrap pt1 pb1 pl2 pr2 white br2 f-supersmall">{tag.name}</span>) : null}
+                    <Tags
```
Comment (aileen) [inline src/components/tutorials/TutorialCard.js:27]: This should result in the exact same output as before, but doesn't atm. Problems are the `visibility` attribute and the `separator` as mentioned above.

File: src/components/common/Tags.js (hunk 3 above)
Comment (ErisDS) [reply src/components/common/Tags.js]: I have no idea why because they aren't missing?

Comment (aileen) [reply src/components/common/Tags.js]: I _think_ it's because it's a separate fn maybe?

File: src/components/common/Tags.js (hunk 2 above)
Comment (rshbhgrg) [reply src/components/common/Tags.js]: Separator issue was fixed in helper in https://github.com/TryGhost/Ghost-SDK/pull/30

File: src/components/common/Tags.js (hunk 1 above)
Comment (rshbhgrg) [reply src/components/common/Tags.js]: visibility attribute works now after fixes to visibility helpers in SDK now, this component should now be used directly as `import { Tags } from '@tryghost/helpers-gatsby'`

**Review by aileen — APPROVED** [review]

**Pull request merged**